### PR TITLE
Add fedrag example with embedding training

### PR DESCRIPTION
**Pull request by ZiyueXu77**
### Description

Add example folder for FedRAG, first example is added for embedding model training

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by ZiyueXu77** [comment]
Fully functional, but with job template, to add python execution

**Comment by ZiyueXu77** [comment]
to use FedJob API, need updates to https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_opt/pt/job_config/model.py to enable persistor args, save for later

**Comment by SYangster** [comment]
> to use FedJob API, need updates to https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_opt/pt/job_config/model.py to enable persistor args, save for later

You can also add the persistor explicitly to the job with the args you need. The PTModel is just a wrapper for convenience.

File: examples/advanced/rag/embedding/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+We can see the "sag_pt_deploy_map" template is available, with which we further generate job configs for embedding model training as:
+```
+nvflare job create -force \
+  -j "/tmp/embed/nvflare/job" -w "sag_pt_deploy_map" -sd "code" \
+  -f meta.conf min_clients=3 \
+  -f app_1/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name nli" \
+  -f app_2/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name squad" \
+  -f app_3/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name quora" \
+  -f app_server/config_fed_server.conf model_class_path="st_model.SenTransModel" components[0].args.model.args.model_name="microsoft/mpnet-base" min_clients=3 num_rounds=7 key_metric="eval_loss" negate_key_metric=True 
+```
```
Comment (YuanTingHsieh) [inline examples/advanced/rag/embedding/README.md:72]: Is it possible to use new Job API for this?

File: examples/advanced/rag/embedding/job_template/sag_pt_deploy_map/app_1/config_fed_client.conf [hunk 2]
```diff
@@ -0,0 +1,91 @@
+{
```
Comment (YuanTingHsieh) [inline examples/advanced/rag/embedding/job_template/sag_pt_deploy_map/app_1/config_fed_client.conf:1]: Do we need this template here?

File: examples/advanced/rag/generation/README.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# RAG with federated context retrieval
+This example shows how to use [NVIDIA FLARE](https://nvidia.github.io/NVFlare) for RAG with federated context retrieval, collecting context from multiple clients for generation tasks.
```
Comment (YuanTingHsieh) [inline examples/advanced/rag/generation/README.md:2]: should we add this in another PR

File: examples/advanced/rag/embedding/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+ TrainData | STSB_pearson_cos | STSB_spearman_euc | NLI_cos_acc | NLI_euc_acc
+--- |------------------|-------------------|-------------| ---
+NLI | 0.7586           | 0.7895            | 0.8033      | 0.8045
+Squad | 0.8206           | 0.8154            | 0.8051      | 0.8042
+Quora | 0.8161           | 0.8121            | 0.7891      | 0.7854
+All | 0.8497           | 0.8523            | 0.8426      | 0.8384
+Federated | 0.8444           | 0.8368            | 0.8269      |  0.8246
+
+As shown, the federated training results are better than individual site's, and can be close to the centralized training results, demonstrating the effectiveness of NVFlare in embedding model tuning tasks.
```
Comment (YuanTingHsieh) [inline examples/advanced/rag/embedding/README.md:94]: Could you add a little more explanation to this paragraph?

1. Include links to "SQUAD", "QUORA" dataset?
2. Explain left hand side, do you mean: 

NLI -> Single site training using NLI data

Squad -> Single site training using Squad data

Quora -> Single site training using Quora data

All -> Single site training using NLI, Squad and Quora data, do you just use them all or there is any ratio to mix

Federated -> Three sites where each site contains its own data of NLI, Squad or Quora

**Review by YuanTingHsieh — COMMENTED** [review]
Great examples, added some suggestions

File: examples/advanced/rag/embedding/README.md (hunk 1 above)
Comment (ZiyueXu77) [reply examples/advanced/rag/embedding/README.md]: Thanks! working on it now

File: examples/advanced/rag/generation/README.md (hunk 3 above)
Comment (ZiyueXu77) [reply examples/advanced/rag/generation/README.md]: removed this part while keeping the folder structure for future additions

File: examples/advanced/rag/embedding/job_template/sag_pt_deploy_map/app_1/config_fed_client.conf (hunk 2 above)
Comment (ZiyueXu77) [reply examples/advanced/rag/embedding/job_template/sag_pt_deploy_map/app_1/config_fed_client.conf:1]: removed the template part

File: examples/advanced/rag/embedding/README.md (hunk 1 above)
Comment (ZiyueXu77) [reply examples/advanced/rag/embedding/README.md]: done and updated

**Review by YuanTingHsieh — APPROVED** [review]
Thanks!LGTM

**Pull request merged**